airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -0,0 +1,129 @@
+      - name: {{ container.name }}
+        image: {{ container.image }}
+        command: {{ container.command }} {% endfor %}
+      restartPolicy: Never
+  backoffLimit: 0
+"""
+
+
+class KubernetesContainerInformation():
+    def __init__(self,
+                 name,
+                 image,
+                 command):
+        self.name = name
+        self.image = image
+        self.command = command
+
+
+class KubernetesJobOperator(BaseOperator):
+    """
+    The KubernetesJobOperator spins up, executes, and cleans up a Kubernetes job.
+    It does so by following these steps:
+        1. Create a Kubernetes Job yaml from job_name and kubernetes_container_information_list
+        2. Instatiate the Job
+        3. Poll and wait for Job completion
+        4. Delete the Job (and Pod)
+
+    # TODO: Expand yaml creation or allow people to pass custom Kubernetes Job yamls.
+            That way we can specify things like: resources, env_vars, volumes, etc.
+
+    :param job_name: meaningful name for the job
+    :type job_name: string
+    :kubernetes_container_information_list: list of containers required to execute the Job
+    :type kubernetes_container_information_list: list of type KubernetesContainerInformation
+    """
+    def __init__(self,
+                 job_name,
+                 kubernetes_container_information_list,
+                 *args,
+                 **kwargs):
+        super(KubernetesJobOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.kubernetes_container_information_list = kubernetes_container_information_list
+
+    def run_subprocess(self, args, stdin=None):
+        try:
+            result = subprocess.Popen(args=args, stdin=stdin, stdout=PIPE)
+            logging.info(result.stdout.read())
+        except Exception as e:
+            raise e
+
+    def clean_up(self):
+        """
+        Deleting the job removes the job and all related pods.
+        Cleans old yamls off Airflow worker.
+        """
+        self.run_subprocess(args=['kubectl', 'delete', 'job', '%s' % self.unique_job_name])
+        self.run_subprocess(args=['rm', '%s' % self.filename])
+
+    def on_kill(self):
+        """
+        Run clean up.
+        Raising an error will fail the KubernetesJobOperator task.
+        """
+        self.clean_up()
+        raise Exception('Job %s was killed.' % self.unique_job_name)
+
+    def poll_job_completion(self):
+        """
+        Polls for Job completion every 60 seconds.
+        Any Failed pods will raise an error and fail the KubernetesJobOperator task.
+        """
+        logging.info('Polling for completion of job: %s' % self.unique_job_name)
+        while True:
+            job_description = subprocess.Popen(args=['kubectl', 'describe', 'job', self.unique_job_name], stdout=PIPE)
+            status_description = subprocess.Popen(args=['grep', 'Pods Statuses:'], stdin=job_description.stdout, stdout=PIPE)
```

Comment:
I think you should eliminate this subprocess, and just execute your `re` directly on the output. It avoids depending on grep and will hopefully be simpler (and it might also be faster which doesn't really matter).

Reply:
I've taken most of this suggestion!